ext/semian/semian.c

```diff
@@ -167,13 +134,11 @@ static VALUE
 update_ticket_count(update_ticket_count_t *tc)
 {
   short delta;
-  struct timespec ts = { 0 };
-  ts.tv_sec = kInternalTimeout;
 
   if (get_max_tickets(tc->sem_id) != tc->tickets) {
     delta = tc->tickets - get_max_tickets(tc->sem_id);
 
-    if (perform_semop(tc->sem_id, kIndexTickets, delta, 0, &ts) == -1) {
+    if (perform_semop(tc->sem_id, kIndexTickets, delta, 0) == -1) {
```

Comment:
Agree with @csfrancis, we need to have retries and a timeout here. This is also needed in https://github.com/Shopify/semian/pull/54 for acquiring the lock around mutating the shared memory. They can share a fallback for `semtimedop` though.